sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -94,6 +94,8 @@ private[sql] class CSVOptions(@transient private val parameters: Map[String, Str
 
   val maxCharsPerColumn = getInt("maxCharsPerColumn", 1000000)
 
+  val maxStoredMalformedPerPartition = getInt("maxStoredMalformedPerPartition", 1)
```

Comment:
Since this affects logging, I suggest a name that implies that. How about `maxLogRecordsPerPartition`